### PR TITLE
Add scrap metal to mattress dissassembly

### LINKED ISSUE #30973: Mattresses give enormous amounts of rags when cut up
# Describe the bug

When cutting up a mattress, it produces exactly 1199 rags.

# Steps To Reproduce
1: Disassemble a bed.
2: Cut up the dropped mattress item.

# Expected behavior

Less rags produced, somewhere between 75 and 125, along with a handful of springs/scrap metal.

# Actual behavior

Way too many rags produced, specifically 1199, no springs, no scrap metal.

# Screenshots
image

# Versions and configuration

 - OS: Windows 10
 - Game Version: 0.D-361
 - Graphics version: Tiles

### PR DESCRIPTION
**Pull request by Feuerhai**
#### Summary
```SUMMARY: Balance "Add scrap metal to mattress dissassembly"```

#### Purpose of change

Afaik most mattresses have some kind of spring in it, but their size, number and the construction of the spring-system differs a lot.
Also fixes #30973

#### Describe the solution
Adds 100 scrap metal to the disassembely result of mattresses. Also adds the steel component, which indirectly prevents the mattress from being cut up instead of dissassembled.

#### Describe alternatives you've considered
Use the spring item instead of scrap metal.
Create a new kind of mattress, a mattress with springs.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
According to my research when making the mattress item, spring mattresses haven't been the most common type of mattress for some time.

I'm also not convince this will fix the issue you reference, since it hasn't done anything to change the rag yield of the mattress. I think the problem is related to the mattress having a high weight and being made of cotton. Most likely you'll now get an enormous amount of rags and scrap metal.

**Comment by I-am-Erk** [comment]
Ah, no, scratch that. Trying to find the old source I found a more specific one, which clarifies that while most new mattresses are foam, innersprings are still the most common, so having scrap metal be part of the equation is fine. I still don't think this is going to solve the issue though.

**Comment by Feuerhai** [comment]
I did some testing with a toolbox and wasn't able to cut up the mattress anymore after adding the steel material, so I thought it would be fine. Regarding smashing a bed, I'll get right to it.

**Comment by I-am-Erk** [comment]
Ah, if that stops cutting up and forces you to go through the disassembly routine then that's a reasonable fix I guess.

We may want the "cut up" code made a little more robust since it seems to break down for larger objects.

File: data/json/uncraft/generic.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "difficulty": 1,
     "time": 2200,
     "qualities": [ { "id": "CUT", "level": 1 } ],
-    "components": [ [ [ "rag", 60 ] ], [ [ "sheet", 4 ] ] ]
+    "components": [ [ [ "rag", 60 ] ], [ [ "sheet", 4 ] ], [ [ "scrap", 100 ] ] ]
```
Comment (I-am-Erk) [inline data/json/uncraft/generic.json]: I'd suggest reducing the scrap a lot, and adding a couple units of wire instead. Maybe 20 scrap and 20 wire? I'm going with the idea that an innerspring could basically be used to make a segment of chain link fence (in the most redneck way imaginable), and it takes 20 wire to make a segment of chainlink fence.

**Comment by Feuerhai** [comment]
Totally forgot about wire, great idea!

**Pull request merged**